mne/io/brainvision/brainvision.py

```diff
@@ -245,7 +248,7 @@ def _read_annotations_brainvision(fname, sfreq='auto'):
         The annotations present in the file.
     """
     onset, duration, description, date_str = _read_vmrk(fname)
-    orig_time = None if date_str == '' else _str_to_meas_date(date_str)
+    orig_time = _str_to_meas_date(date_str)
```

Comment:
annotations `orig_time` has to either be `None` or a date. If you pass 'foobar' or '' some tests should be breaking. If its not the case I did not do a good enough job on testing it.

Reply:
I would be fine if `_str_to_emas_date` returns `None` when needed. But I think you are doing the oposite.

Reply:
> annotations orig_time has to either be None or a date

yes, +1

> If you pass 'foobar' or '' some tests should be breaking.

yes, +1, this is taken care of in `_str_to_meas_date`, see: `meas_date = datetime.strptime(date_str, '%Y%m%d%H%M%S%f')` part (raise if format error)

> I would be fine if _str_to_emas_date returns None when needed. But I think you are doing the oposite.

No, I am doing exactly that: `_str_to_meas_date` now either returns `None` or a date

What I do here is fix an error - see this line:

`orig_time = None if date_str == '' else _str_to_meas_date(date_str)`

According to how `_read_vmrk` was defined above, `date_str` would actually NEVER be `''` ... it would be either `None`, or a date string ... then if `date_str` is `None`, `_str_to_meas_date` throws an error, because it tries to do `date_str.strip()`, which is not defined for NoneTypes

With my PR we:
- avoid running into `None.strip()` error
- get a better docstring on what `_read_vmrk` actually returns
- get rid of an unnecessary (and impossible to reach) conditional (line 248)
- get a second fix, see line 213
  - prior to this PR, we would have potentially equated `date_str == this_duration`, because we did not check that the info data is actually of length 5 (mtype, mdesc, this_onset, this_duration ... AND date_str) 
  - it could very well be that we get a marker of mtype `NewSegment`, but without date information (=length 4) ... then the `info.split(',')[-1]` would correspond to `this_duration`